PR: Fix legal texts and make them consistent across all files

CAM-Gerlach opened the pull request:
Makes a number of small changes to correct errors/omissions, improve consistency, and clarify the legal language. Specifically:

* Adds an ``AUTHORS.txt`` file like our other repos have to clarify the meaning of "Spyder Kernels Contributors" and credit the appropriate people
* Adds the license text in ``__init__.py`` as in main Spyder that all the file headers refer to, but is not actually present
* Updates the file legal headers to one, consistent standard (several different formattings were used, and they all reference one of several different, incorrect paths for the __init__.py file since they were copied from other projects; also, makes the copyright statement up to date and legally valid

ccordoba12 commented on AUTHORS.txt:
```diff
@@ -0,0 +1,5 @@
+The Spyder Kernels Contributors are composed of:
+
+* Carlos Cordoba <ccordoba12@gmail.com> (Spyder-Kernels creator and maintainer)
+* All other developers that have contributed to the spyder-kernels repository:
```
This is wrong. It needs to be

    All other developers that have contributed to the Spyder repository

because I didn't take renames when I got the git history of the files took from the Spyder repo (my mistake, but I realized too late).

For example, Pierre is not here, but he was the one that created most of the files present here.

CAM-Gerlach replied:
...

First I don't push my changes explicitly since

> I hadn't yet addressed your other review comment :-)

Then, since you commented

> But it seems you forgot to push your local changes :-)

I push them with the above disclaimer

and now I'm told

> This is still missing. Please address it.

I just can't win, can I jeje ¯\_(ツ)_/¯

CAM-Gerlach replied:
TL;DR: I'm still researching the commit history on the relevant files to see what can be done.

CAM-Gerlach replied:
@ccordoba12 Tis done!

CAM-Gerlach commented:
> Could you set only the initial year in the Copyright statement?

Yeah; technically by [best practices](https://www.copyrightlaws.com/copyright-notice-year/), the GNU requirements, etc the year the work was last changed should be included and updated when the file is, but its not *that* important and can just be included in the __init__ and LICENSE files. Modified as requested.

> Also, please notice that the copyright goes from 2009, the initial Spyder release date.

As mentioned above, I was mislead by the lack of renames. Fixed.

ccordoba12 commented:
> I was mislead by the lack of renames. Fixed.

Sure, no problem. But it seems you forgot to push your local changes :-)

ccordoba12 commented on LICENSE.txt:
```diff
@@ -1,6 +1,6 @@
 MIT License
 
-Copyright (c) 2018 Spyder IDE
+Copyright (c) 2009-2018 Spyder Kernels Contributors (see AUTHORS.txt)
```
Yes, it is. Please remove it, the less work we have to do the better.

CAM-Gerlach replied:
Done...

CAM-Gerlach commented:
@ccordoba12 Finally all done with linking the previous versions of the files and their contributors, and double-checking through the commit history of each to ensure everyone was included. Therefore, all your feedback *should* now be addressed. Thanks.
